test/client/spec/victory-util/collection.spec.js

```diff
@@ -174,4 +177,71 @@ describe("collections", () => {
       expect(Collection.allSetsEqual(comparisons)).to.eql(false);
     });
   });
+
+
+  describe("areVictoryPropsEqual", () => {
+    let sandbox;
+    beforeEach(() => {
+      sandbox = sinon.sandbox.create();
+      sandbox.spy(Collection, "checkEquality");
+    });
+    afterEach(() => {
+      sandbox.restore();
+    });
+
+    it("returns early when nested collections are strictly equal", () => {
+      const a = { test: { nested: "a" } };
+      const b = a;
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+      expect(Collection.checkEquality).calledOnce.and.returned(true);
+    });
+
+    it("returns early when nested collections are not the same length", () => {
+      const a = { test: { nested: "a" }, test2: { nested: "b" } };
+      const b = { test: { nested: "a" } };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(false);
+      expect(Collection.checkEquality).calledOnce.and.returned(false);
+    });
+
+    it("returns early if values are not the same type", () => {
+      const a = { test: { nested: "a" } };
+      const b = { test: () => "a" };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(false);
+      expect(Collection.checkEquality).calledOnce.and.returned(false);
+    });
+
+    it("returns early when nested elements are empty", () => {
+      const a = { a: [] };
+      const b = { a: [] };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+      expect(Collection.checkEquality).calledOnce.and.returned(true);
+    });
+
+    it("returns true if values are functions", () => {
+      const a = { test: () => "a" };
+      const b = { test: () => "a" };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+    });
+
+    it("recursively checks equality for nested collections", () => {
+      const a = { test: { nested: "a" }, test2: { nested: "a" } };
+      const b = { test: { nested: "a" }, test2: { nested: "a" } };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+      expect(Collection.checkEquality).calledThrice;
+    });
+
+    it("does not recursively check date objects", () => {
+      const a = { test: new Date(), test2: new Date() };
+      const b = { test: new Date(), test2: new Date() };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+      expect(Collection.checkEquality).calledOnce.and.returned(true);
+    });
+
+    it("returns equal for equivalent date objects", () => {
+      const a = { test: new Date(2010, 1, 1) };
+      const b = { test: new Date("Mon Feb 01 2010 00:00:00 GMT-0800 (PST)") };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
+    });
+
+  });
```

Comment:
Looking good generally, but I think we've got some corner cases to cover still. For example, we return `true` for `{}` vs. `[]` when that probably should be `false`. Here's a diff with that test and a whole bunch of lodash "isEqual" unit tests (_note_ we should not just copy them like in my diff, but use them as a starting point for our tests).

I think we'll want to look at this diff, and additional lodash isEqual unit tests for additional cases we should cover: https://github.com/lodash/lodash/blob/4.17.4/test/test.js#L9477

```diff
diff --git a/test/client/spec/victory-util/collection.spec.js b/test/client/spec/victory-util/collection.spec.js
index eeab6d4..4fa14d2 100644
--- a/test/client/spec/victory-util/collection.spec.js
+++ b/test/client/spec/victory-util/collection.spec.js
@@ -179,10 +179,12 @@ describe("collections", () => {
   });
 
 
-  describe("areVictoryPropsEqual", () => {
+  describe.only("areVictoryPropsEqual", () => {
     let sandbox;
     beforeEach(() => {
-      sandbox = sinon.sandbox.create();
+      sandbox = sinon.sandbox.create({
+        useFakeTimers: true
+      });
       sandbox.spy(Collection, "checkEquality");
     });
     afterEach(() => {
@@ -243,5 +245,39 @@ describe("collections", () => {
       expect(Collection.areVictoryPropsEqual(a, b)).to.equal(true);
     });
 
+    // FAILS
+    it("returns false for empty object vs. array", () => {
+      const a = { a: [] };
+      const b = { a: {} };
+      expect(Collection.areVictoryPropsEqual(a, b)).to.equal(false);
+      expect(Collection.checkEquality).calledOnce.and.returned(false);
+    });
+
+    // Disable symbol tests.
+    // https://github.com/lodash/lodash/blob/4.17.4/test/test.js#L9480-L9496
+    const symbol1 = true;
+    const symbol2 = false;
+    const pairs = [
+      [1, 1, true], [1, Object(1), true], [1, '1', false], [1, 2, false],
+      [-0, -0, true], [0, 0, true], [0, Object(0), true], [Object(0), Object(0), true], [-0, 0, true], [0, '0', false], [0, null, false],
+      [NaN, NaN, true], [NaN, Object(NaN), true], [Object(NaN), Object(NaN), true], [NaN, 'a', false], [NaN, Infinity, false],
+      ['a', 'a', true], ['a', Object('a'), true], [Object('a'), Object('a'), true], ['a', 'b', false], ['a', ['a'], false],
+      [true, true, true], [true, Object(true), true], [Object(true), Object(true), true], [true, 1, false], [true, 'a', false],
+      [false, false, true], [false, Object(false), true], [Object(false), Object(false), true], [false, 0, false], [false, '', false],
+      [symbol1, symbol1, true], [symbol1, Object(symbol1), true], [Object(symbol1), Object(symbol1), true], [symbol1, symbol2, false],
+      [null, null, true], [null, undefined, false], [null, {}, false], [null, '', false],
+      [undefined, undefined, true], [undefined, null, false], [undefined, '', false]
+    ];
+
+    // SOME FAIL
+    pairs.forEach((vals) => {
+      it(`lodash primitive comparison: ${vals}`, () => {
+        const a = vals[0];
+        const b = vals[1];
+        const expected = vals[2];
+        expect(Collection.areVictoryPropsEqual(a, b)).to.equal(expected);
+      });
+    });
+
   });
 });
```